app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,52 @@
+module UserOmniauthConcern
```

Comment:
I think this concern is not properly structured for multiple providers in future.